packages/components/upload/__tests__/upload.test.tsx

```diff
@@ -131,6 +131,42 @@ describe('<upload />', () => {
       expect(onError).toHaveBeenCalled()
     })
 
+    // Ensure that the modified data in before-upload can be correctly passed into the upload request. (#12029)
+    test('in beforeUpload change data correctly to request', async () => {
+      const keyList: string[] = []
+      const beforeUpload = vi.fn((file) => (data.value.key = file.name))
+      const httpRequest = ref(
+        vi.fn((val) => {
+          keyList.push(val?.data?.key)
+          return Promise.resolve()
+        })
+      )
```

Comment:
The `ref(vi.fn(...))` have any special significance?

Reply:
It is just a copy of the previous case transformation.

Reply:
IMO, `ref()` is redundant here

Reply:
Ok, then I will delete the others later.

Reply:
done.